pangeo/pbs.py

```diff
@@ -0,0 +1,236 @@
+from contextlib import contextmanager
+import logging
+import os
+import socket
+import subprocess
+import sys
+import toolz
+
+from distributed import LocalCluster
+from distributed.utils import (tmpfile, get_ip_interface, format_bytes,
+        parse_bytes, ignoring)
+
+template = """
+#!/bin/bash
+
+#PBS -N %(name)s
+#PBS -q %(queue)s
+#PBS -A %(project)s
+#PBS -l %(resource_spec)s
+#PBS -l walltime=%(walltime)s
+#PBS -j oe
+#PBS -m abe
+
+%(base_path)s/dask-worker %(scheduler)s \
+    --nthreads %(threads_per_worker)d \
+    --nprocs %(processes)s \
+    --memory-limit %(memory)s \
+    --name %(name)s-%(n)d \
+     %(extra)s
+""".lstrip()
```

Comment:
In the future, we may want to make this template a attribute on the PBSCluster object. For use on Cheyenne, this will certainly work so I'm not suggesting changing this until we want to be entirely machine agnostic.

Reply:
Agreed on both counts.